### PR TITLE
Add Palette GUI controls

### PR DESCRIPTION
**Pull request by ThomasJClark**

### CODE REVIEW

**Comment by ThomasJClark** [comment]
This PR only includes the classes for the GUI controls, the `main` function hasn't been updated to make use of them.  See [this](https://github.com/ThomasJClark/GRIP/tree/palette-gui-demo) branch for a quick demo.

File: src/main/java/edu/wpi/grip/core/operations/PythonScriptOperation.java [hunk 1]
```diff
@@ -128,6 +129,11 @@ public String getDescription() {
         return this.description.getString();
     }
 
+    @Override
+    public java.util.Optional<InputStream> getIcon() {
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/PythonScriptOperation.java]: This is fine. But probably not necessary.

File: src/main/java/edu/wpi/grip/core/operations/PythonScriptOperation.java [hunk 2]
```diff
@@ -128,6 +129,11 @@ public String getDescription() {
         return this.description.getString();
     }
 
+    @Override
+    public Optional<InputStream> getIcon() {
+        return java.util.Optional.of(getClass().getResourceAsStream("/edu/wpi/grip/ui/icons/python.png"));
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/PythonScriptOperation.java]: Same problem here

File: src/main/resources/edu/wpi/grip/ui/MainWindow.fxml.orig [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+<?import javafx.scene.control.*?>
+<?import javafx.scene.layout.*?>
+<?import java.net.URL?>
+<<<<<<< HEAD
+<VBox fx:controller="edu.wpi.grip.ui.controllers.MainWindowController" maxHeight="-Infinity" maxWidth="-Infinity"
+      minHeight="-Infinity"
+=======
+<VBox maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity"
+>>>>>>> 6e9d28f... Add OperationView and PaletteView
```
Comment (JLLeitschuh) [inline src/main/resources/edu/wpi/grip/ui/MainWindow.fxml.orig]: You have a merge conflict that is unresolved here.

File: src/main/resources/edu/wpi/grip/ui/MainWindow.fxml.orig [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+<<<<<<< HEAD
+                <ScrollPane fx:id="pipelineContainer"/>
+=======
+                <FlowPane styleClass="placeholder">
+                    <children>
+                        <Label alignment="CENTER" styleClass="placeholder"
+                               text="Pipeline" textAlignment="CENTER" wrapText="false"/>
+                    </children>
+                </FlowPane>
+>>>>>>> 6e9d28f... Add OperationView and PaletteView
```
Comment (JLLeitschuh) [inline src/main/resources/edu/wpi/grip/ui/MainWindow.fxml.orig]: You have another unresolved merge conflict here

**Comment by JLLeitschuh** [comment]
At the very least we should really have a test that loads up the GUI on travis. That way merge conflicts don't end up in the fxml files.
That's something I can work on if I have some free time.

**Pull request merged**